docs/policy/topology-aware.md

```diff
@@ -300,6 +306,21 @@ Topology hint generation is globally enabled by default. Therefore, using the
 Pod annotation as opt in only has an effect when the whole pod is annotated to
 opt out from hint-aware pool selection.
 
+### Topological Co-location for Pods and Namespaces
+
+The `ColocatePods` or `ColocateNamespaces` configuration options control whether
+the policy will try to co-locate, IOW allocate topologically close to each other
+containers within the same Pod or K8s namespace. Both of these options are opt-in,
+they are off by default. The actual co-location/allocation strategy is implemented
```

Comment:
Suggesting simplifying, switching from implementation to user point of view, and offering direct configuration words "true"/"false" instead of "opt-in"/"off":

Original:
_Both of these options are opt-in, they are off by default. The actual co-location/allocation strategy is implemented by simply injecting implicit affinities to containers which cause them to have affinity for all other containers in the same Pod or namespace._

_Any container that has explicitly annotated affinities by the user will be excluded from annotation injection by these options. Note that since annotations injected to other containers can still be evaluated against these containers, extra care must be taken if explicit annotations are also used while either or both of these options are enabled. Both of these implicitly injected affinities use a weight of 10._

A suggestion:
Both of these options are `false` by default. Setting to `true` is a shorthand for adding affinities of weight 10 to containers in same pods or namespaces.

Containers with user-defined affinities are not extended with colocation affinities. However, colocation affinities of other containers affect the placement of these containers, too. Therefore mixing user-defined and colocation affinities requires carefully thought on affinity weights.

Reply:
> _ ... Note that since annotations injected to other containers can still be evaluated against these containers, extra care must be taken if explicit annotations are also used while either or both of these options are enabled. Both of these implicitly injected affinities use a weight of 10._
> 
> A suggestion: ... However, colocation affinities of other containers affect the placement of these containers, too. Therefore mixing user-defined and colocation affinities requires carefully thought on affinity weights.

Point taken, I'll try to rephrase. But can't do it like that, because that statement is very different from the original one, and also false.

Reply:
Rephrased and updated.